src/jit/codegencommon.cpp

```diff
@@ -4888,6 +4888,82 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
 #pragma warning(pop)
 #endif
 
+
+/*****************************************************************************
+* If any Vector3 args are on stack and they are not pass-by-ref, the upper 32bits
+* must be cleared to zeroes. The native compiler doesn't clear the upper bits
+* and there is no way to know if the caller is native or not. So, the upper
+* 32 bits of Vector argument on stack are always cleared to zero.
+*/
+#ifdef FEATURE_SIMD
+void    CodeGen::genClearStackVec3ArgUpperBits()
+{
+#ifdef DEBUG
+    if (verbose)
+        printf("*************** In genClearStackVec3ArgUpperBits()\n");
+#endif
+
+#if !defined(_TARGET_XARCH_)
+    return;
+#endif // !_TARGET_XARCH_
```

Comment:
In looking at this further, this approach will add extra code in either the case where the parameter is passed in a register and then homed to the stack or vice versa. To generate better code, I think that the appropriate clearing should be done for incoming register args in `genFnPrologCalleeRegArgs()` and for incoming stack args in `genEnregisterIncomingStackArgs()` (though handling the stack->stack case there is a little weird, it seems the best place to deal with it). That way, the zeroing of the upper bits can be done along with the code to appropriately "home" the argument.